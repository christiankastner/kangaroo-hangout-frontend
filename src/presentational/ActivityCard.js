import React from 'react'
import {Card, Icon, Button} from 'semantic-ui-react'

const ActivityCard = props => {
    const {image, name, types, icon, formatted_address} = props.activity
    // const formatTypes = types.reduce((total, type) => total + " " + type)
    const extra = () => {
        return(
            <>
                {props.handleAdd ? <Button data-testid="activity-button" primary onClick={() => props.handleAdd(props.activity)}><Icon name="plus" />Add</Button> 
                : <Button data-testid="activity-button" color="red" onClick={() => props.handleRemove(props.activity)}><Icon name="undo"/>Remove</Button>}
                <Button data-testid="activity-select" secondary onClick={() => props.handleSelectPlace(props.activity)}>Inspect</Button>
            </>
        )
    }

    return (
        <div data-testid="activity-card">
            <Card
                header={name}
                description={<span><img className="icon" src={icon}/>  {formatted_address}</span>}
                extra={extra()}
            />
        </div>
    )
}

ActivityCard.defaultProps = {
    activity: {
        name: "Cannot Find Name",
        icon: <Icon name="marker" />,
        types: "misc",
        formatted_address: "No Result Found"
    }
}

export default ActivityCard