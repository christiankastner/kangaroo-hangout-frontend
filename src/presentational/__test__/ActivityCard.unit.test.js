import React from 'react'
import ActivityCard from "./../ActivityCard"
import { render, cleanup } from '@testing-library/react'
import "@testing-library/jest-dom/extend-expect"

import renderer from "react-test-renderer"

describe("Activity Card", () => {
    afterEach(cleanup);
    
    it("Renders card correctly with no props", () => {
        const { getByTestId, getByText } = render(<ActivityCard/>)

        expect(getByTestId("activity-card")).toHaveTextContent("Cannot Find Name")
        expect(getByTestId("activity-card")).toHaveTextContent("No Result Found")
    })

    it("Renders card correctly with props", () => {
        const activity = {name: "Joe's Crab Shack", formatted_address: "1600 Pennsylvania Avenue"}
        const { getByTestId } = render(<ActivityCard activity={activity}></ActivityCard>)
        expect(getByTestId("activity-card")).toHaveTextContent("Joe's Crab Shack")
        expect(getByTestId("activity-card")).toHaveTextContent("1600 Pennsylvania Avenue")
    })

    it("Renders add button with the right prop", () => {
        const addCallback = () => {
            return 
        }
        const activity = {name: "Joe's Crab Shack", formatted_address: "1600 Pennsylvania Avenue"}
        const { getByTestId } = render(<ActivityCard handleAdd={addCallback} activity={activity}></ActivityCard>)
        expect(getByTestId("activity-button")).toHaveTextContent("Add")
    })

    it("Matches snapshot with no props", () => {
        const tree = renderer.create(<ActivityCard></ActivityCard>).toJSON()
        expect(tree).toMatchSnapshot()
    })
    it("Matches snapshot with props", () => {
        const activity = {name: "Joe's Crab Shack", formatted_address: "1600 Pennsylvania Avenue"}
        const tree = renderer.create(<ActivityCard activity={activity}></ActivityCard>).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
