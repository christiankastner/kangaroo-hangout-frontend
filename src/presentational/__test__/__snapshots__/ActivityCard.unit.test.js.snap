// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity Card Matches snapshot with no props 1`] = `
<div
  data-testid="activity-card"
>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Cannot Find Name
      </div>
      <span>
        <img
          className="icon"
          src={
            <Icon
              as="i"
              name="marker"
            />
          }
        />
          
        No Result Found
      </span>
    </div>
    <div
      className="extra content"
    >
      <button
        className="ui red button"
        data-testid="activity-button"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="undo icon"
          onClick={[Function]}
        />
        Remove
      </button>
      <button
        className="ui secondary button"
        data-testid="activity-select"
        onClick={[Function]}
      >
        Inspect
      </button>
    </div>
  </div>
</div>
`;

exports[`Activity Card Matches snapshot with props 1`] = `
<div
  data-testid="activity-card"
>
  <div
    className="ui card"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        Joe's Crab Shack
      </div>
      <span>
        <img
          className="icon"
        />
          
        1600 Pennsylvania Avenue
      </span>
    </div>
    <div
      className="extra content"
    >
      <button
        className="ui red button"
        data-testid="activity-button"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="undo icon"
          onClick={[Function]}
        />
        Remove
      </button>
      <button
        className="ui secondary button"
        data-testid="activity-select"
        onClick={[Function]}
      >
        Inspect
      </button>
    </div>
  </div>
</div>
`;
